core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -160,11 +161,14 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
 
   override def preStart(): Unit = {
     super.preStart()
+
     consumer = settings.createKafkaConsumer()
+    currentPollTask = schedulePollTask()
   }
 
   override def postStop(): Unit = {
-    pollTask.cancel()
+    if (currentPollTask != null)
+      currentPollTask.cancel()
```

Comment:
Nice :-) Hadn't thought of that. I've implemented your suggested solution. `this` would be different across restarts as opposed to `self`, right?